services/registry/src/sync/sync-service.ts

```diff
@@ -1,60 +1,63 @@
 import sources from "../sources-config";
 import { defer, from, empty, Observable } from "rxjs";
 import {
-  mergeMap,
-  concat,
-  delay,
-  repeat,
-  map,
-  scan,
-  shareReplay,
-  distinctUntilChanged,
-  tap,
-  filter,
-  catchError,
-  combineAll
+    mergeMap,
+    concat,
+    delay,
+    repeat,
+    map,
+    scan,
+    shareReplay,
+    distinctUntilChanged,
+    catchError,
+    combineAll,
+    startWith,
+    filter,
 } from "rxjs/operators";
 import {
-  makeGqlDocumentFromGqlSources,
-  GqlSources
+    makeGqlDocumentFromGqlSources,
+    GqlSources
 } from "../graphql/create-schema";
 import { print } from "graphql/language/printer";
 
 const emitAndWait = (duration: number) => concat(empty().pipe(delay(duration)));
 
 export const schemas$: Observable<GqlSources> = from(
-  Object.entries(sources)
+    Object.entries(sources)
 ).pipe(
-  map(([sourceName, source]) =>
-    defer(() => source.getSchemas()).pipe(
-      mergeMap(schemaByName => from(Object.entries(schemaByName))),
-      map(([name, schema]) => [`${sourceName}.${name}`, schema]),
-      emitAndWait(5000) as any,
-      catchError(err => {
-        console.warn("Error getting schema from source", source, err);
-        return empty();
-      }),
-      repeat(),
-      scan(
-        (schemaBySources: GqlSources, [source, schema]: [string, string]) =>
-          ({
-            ...schemaBySources,
-            [source]: schema
-          } as GqlSources),
-        {}
-      )
-    )
-  ),
-  combineAll(),
-  map(schemas => schemas.reduce((acc, next) => ({ ...acc, ...next }))),
-  shareReplay(1)
+    map(([sourceName, source]) =>
+        // FIXME: Not only schemas
+        defer(() => source.getGqlObjects("gqlschemas")).pipe(
```

Comment:
Let's change the getGqlObject to return all source data. (no "kind" parameter).
After we get the combined RawConfigurationData, we can apply several field transformers (schema, endpoints, etc...) to create the final GqlConfigurationMessage